sklearn/metrics/_classification.py

```diff
@@ -3222,11 +3238,27 @@ def brier_score_loss(y_true, y_prob, *, sample_weight=None, pos_label=None):
     >>> brier_score_loss(y_true, np.array(y_prob) > 0.5)
     0.0
     """
+    # TODO(1.7): remove in 1.7 and reset y_proba to be required
+    if y_proba is not None and str(y_prob) != "deprecated":
```

Comment:
There is an edge case where an array like has `str(y_pred)=="deprecated"`. I think it's safer to check for str first:

```suggestion
    if y_proba is not None and (isinstance(y_prob, str) and y_prob == "deprecated"):
```